inst/include/frogger.hpp

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <iostream>
+#include "types.hpp"
+
+const int MALE = 0;
+const int FEMALE = 1;
+
+template <typename real_type>
+State<real_type> model_runner(int time_steps,
+                              const Parameters<real_type>& pars) {
+  State<double> state(pars.age_groups_pop, pars.num_genders);
+  initialise_model_state(pars, state);
+  auto state_next = state;
+  for (int time = 1; time <= time_steps; ++time) {
```

Comment:
With discrete time models, it's always worth getting pretty concrete about what `time` means here - is this the value at the beginning or at the end of the step?

Reply:
It's the mid point of the year, I've added a comment for that

Reply:
Oh yeah this is a bit confusing, I tried to comment it to explain in `frogger.cpp` where we set the `max_sim_years`. But probably I should put something in here too.

The initial state is filled in from input data at time index 0 (represents year 1970 in example data). In the loop, at each time step we fill `state_next` from `state` and the input data (e.g. `survival`, `age_sex_fertility_ratio` etc.) at time `time`. So loop runs from 1 as we have used the first years data for initial state. 

I'm then running up to `<= time_steps` because I wanted the `sim_years` arg to `run_base_model` to be the number of times the loop runs. The logic which sets `time_steps` sets max value to be 1 less than the length so we don't run off the end of the input data.